### PR TITLE
update README.md

### PR DESCRIPTION
**Pull request by figot**
Translation Plan-models简介-汉译英 #723
https://github.com/PaddlePaddle/models/issues/723

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,18 +70,65 @@ Sequence-to-sequence model has a wide range of applications. This includes machi
...
+
+In the speech recognition task, we provide a complete pipeline based on the DeepSpeech2 model, including: feature extraction, data enhancement, model training, language model, decoding module, etc. At the same time, we provide a trained model and experience example. Everyone can use their own Voice to experience the fun of speech recognition.
+
+14.1 [Speech Recognize: DeepSpeech2](https://github.com/PaddlePaddle/DeepSpeech)
+
 
 This tutorial is contributed by [PaddlePaddle](https://github.com/PaddlePaddle/Paddle) and licensed under the [Apache-2.0 license](LICENSE).
```
Comment (skylarch) [inline README.md:134]: # Introduction of the Models

[Documentation Status](https://github.com/PaddlePaddle/models)
[Documentation Status](https://github.com/PaddlePaddle/models)
[License](LICENSE)

PaddlePaddle provides a wealth of computing units to help you build diverse deep learning models in a modular way and solve different application problems. Here, we provide different neural network models for everyone to learn and use for common machine learning tasks.


## 1. Word Vectors

A word vector represents a word with a real vector. Each dimension of the vector represents a potential grammatical or semantic feature of the text. It is one of the most successful concepts and results of deep learning applied in the field of natural language processing. In a broad sense, word vectors can also be applied to ordinary discrete features. The learning of word vectors is usually an unsupervised learning process; therefore, we can make full use of the massive unmarked data to capture the relationships between features and effectively solve the problems like feature sparseness, tag information missing, and data noise. However, in a common word vector learning method, the last layer of the model often encounters a very large-scale classification problem, which is a bottleneck for computational performance.

In these following word vector tasks, we show how to use Hierarchical-Sigmoid and Noise Contrastive Estimation (NCE) to accelerate the learning of word vectors.

- 1.1 [Hsigmoid Accelerated Word Vector Training] (https://github.com/PaddlePaddle/models/tree/develop/hsigmoid)
- 1.2 [Accelerating the Training Process of a Language Model with Noise Contrast Estimation] (https://github.com/PaddlePaddle/models/tree/develop/nce_cost)


## 2. RNN Language Models

A language model is an important basic model in the field of natural language processing. Besides getting the word vectors (byproducts of language model training), it can also help us generate texts. Given a few words, the language model helps us predict the word most likely to come next.

In a task that uses the language model to generate texts, we will focus on the recurrent neural network language model. Everyone can quickly adapt to their own training corpus following the instructions in the document and construct interesting models that can do tasks such as automatic poetry composition and automatic prose composition.

- 2.1 [Generating Text using a Recurrent Neural Network Language Model] (https://github.com/PaddlePaddle/models/tree/develop/generate_sequence_by_rnn_lm)

## 3. Click Rate Prediction

The click rate prediction model predicts the probability that a user clicks on an ad. It is one of the core algorithms of the technology relating to advertising. Logistic regression has a good ability to learn large-scale sparse features and dominates the early development of CTR prediction tasks. In recent years, DNN models gradually become the main force in CTR estimation tasks due to its strong learning ability.

In a CTR estimation task, we first introduce the Wide & Deep model proposed by Google. This model combines DNN's suitability for learning abstract features and logistic regression's advantages for large-scale sparse features. It can be used as a relatively mature model framework and has some application in the industry. At the same time, we provide a deep neural network model based on a factorization machine that fuses a factorization engine and a deep neural network that respectively model low-order interactions and high-order interactions between input attributes.

- 3.1 [Wide & Deep CTR Prediction Model] (https://github.com/PaddlePaddle/models/tree/develop/ctr/README.cn.md)
- 3.2 [Depth Factor Decomposition Engine Based CTR Prediction Model] (https://github.com/PaddlePaddle/models/tree/develop/deep_fm)

## 4. Text Classification

Text classification is one of the most basic tasks in the field of natural language processing. Deep learning can eliminate the complicated feature engineering and use the original text directly as input, which data-drivenly optimize the classification accuracy.

In text categorization tasks, we provide a non-sequential text categorization model based on DNN and a CNN-based sequence model for everyone to learn and use. (For LSTM-based models, see PaddLeBook [Sentiment Classification] (http (www.paddlepaddle.org/docs/develop/book/06.understand_sentiment/index.cn.html) Lesson 1).

- 4.1 [Sentiment Classification Based on DNN/CNN] (https://github.com/PaddlePaddle/models/tree/develop/text_classification)
- 4.2 [Text Classification Based on Double Sequence] (https://github.com/PaddlePaddle/models/tree/develop/nested_sequence/text_classification)

## 5. Learning to Rank

Learning to rank (LTR) is one of the core issues in the research of information retrieval and search engine. It learns a scoring function with machine learning methods to score candidates for ranking and then determines the order relationship according to the scores. Deep neural networks are used to model the score function and constitute various deep learning LTR models.

In this learning to rank task, we introduce the pairwise ranking model based on RankLoss loss function and the listwise ranking model based on LambdaRank loss function. (See Pointwise learning strategy in the PaddleBook [Recommended System] (http://www.paddlepaddle.org/docs/develop/ Book/05.recommender_system/index.cn.html) Lesson 1).

- 5.1 [Pairwise and Listwise Learning to Rank Model] (https://github.com/PaddlePaddle/models/tree/develop/ltr)

## 6. Structured Semantic Model

Deeply structured semantic model is a neural network-based semantic matching model framework, which can be used to learn the semantic similarity between two information entities or texts. DSSM uses DNN, CNN, or RNN to map two information entities or texts into the same continuous low latitude semantic space. In this semantic space, two entities or texts can be represented at the same time. Then, we define distance metrics and matching functions to describe and learn the semantic similarities of the entities or texts in the same semantic space.

In this structured semantic model task, we demonstrate how to model the semantic similarity between two strings. The model supports different network structures such as DNN (deep neural network), CNN (convolutional neural network, or ConvNet), and RNN (recursive neural network), as well as different loss functions such as classification, regression, and ordering. This example uses the simplest text data as input and can be used in real-world scenarios by replacing its default training and forecasting data.

- 6.1 [Deeply Structured Semantic Model] (https://github.com/PaddlePaddle/models/tree/develop/dssm/README.cn.md)

## 7. Named Entity Recognition

Given the input sequence, the sequence labeling model attaches a category label to each element in the sequence, which is one of the most basic tasks in the field of natural language processing. With the continuous development of deep learning methods, the circulant neural network is used to learn the feature representation of the input sequence. The conditional random field (CRF) completes the sequence annotation task on the basis of features and gradually becomes the standard solution to the sequence labeling problem.

In this sequence labeling task, we use the named entity recognition (NER) task as an example to introduce how to train an end-to-end sequence labeling model.

- 7.1 [Named Entity Recognition] (https://github.com/PaddlePaddle/models/tree/develop/sequence_tagging_for_ner)

## 8. Sequence-to-Sequence learning

Sequence-to-sequence learning implements mapping between two or even multiple indefinite-lengthed models and has a wide range of applications, such as machine translation, intelligent dialogue and Q & A, creative corpus creation, automatic coding (eg, financial portrait coding), and judgment of semantic correlation between text strings, etc.

In sequence-to-sequence learning tasks, we use a machine translation task as an example and provide various enhanced models for everyone to learn and use. Models include a sequence-to-sequence mapping model without attention mechanism which is the basis of all sequence-to-sequence learning models, a model using scheduled sampling to improve the error accumulation problem of RNN model in generation tasks, a neural machine translation model with external memory mechanism, etc. By enhancing the neural network's memory ability, we complete complex sequence-to-sequence learning tasks. In addition to machine translation tasks, we also provide a model based on the deep LSTM network to generate ancient Chinese poetry.

- 8.1 [Neural Machine Translation without Attention Mechanism] (https://github.com/PaddlePaddle/models/tree/develop/nmt_without_attention/README.cn.md)
- 8.2 [Scheduled Sampling] (https://github.com/PaddlePaddle/models/tree/develop/scheduled_sampling)
- 8.3 [Neural Machine Translation with External Memory Mechanism] (https://github.com/PaddlePaddle/models/tree/develop/mt_with_external_memory)
- 8.4 [Chinese Ancient Poetry Generation] (https://github.com/PaddlePaddle/models/tree/develop/generate_chinese_poetry)

## 9. Reading Comprehension

As deep learning and various new technologies continue to push forward the field of natural language processing, we cannot help but ask: How should we confirm that the model truly comprehends human-specific natural language and has a certain ability to understand and reason? Some classic issues in the field of NLP include lexical analysis, syntactic analysis, emotional classification, writing poetry, etc. The classic solution to these problems has some distance from language comprehension from the technical principle point of view. In order to measure the gap between the existing NLP technology and the ultimate goal of language comprehension, we need a task that is difficult, quantifiable, and reproducible. This is also the original intention of reading comprehension. Although the current research status indicates that the models that perform well on the current reading comprehension dataset do not achieve true language comprehension, reading comprehension is still an important task for the test model to understand the language.

Reading comprehension is essentially a kind of automatic questioning and answering. In this task, after the model “reads” a paragraph and answers a given question, we introduce the use of "Learning to Search" method, which transforms reading comprehension task into a task of finding the sentence containing the answer from a paragraph. Ultimately, the model finds where the answer starts and ends in the sentence.

- 9.1 [Globally Normalized Reader] (https://github.com/PaddlePaddle/models/tree/develop/globally_normalized_reader)

## 10. Auto Question Answering

The Question Answering system uses computers to automatically answer the questions raised by users. A question answering task is important for verifying whether the machine has natural language understanding ability. Its research history can be traced back to the origin of artificial intelligence. Compared with the retrieval system, the automatic question answering system is an advanced form of information service. The system no longer returns to the user the sorted keyword-based retrieval results but the accurate natural language answers.

In an automated question-and-answer task, we introduce an end-to-end question-answering system based on deep learning, which translates automated question answering into a sequence annotation problem. The end-to-end question-answering system attempts to build a joint learning model by learning from high-quality "question-evidence-answer" data. At the same time, it learns from knowledge bases, question sentences, and the semantic mapping relationship between semantic representations of corpora. The complex steps of semantic parsing, text retrieval, answer extraction and generation of traditional questions are transformed into a learnable process.

- 10.1 [Fact-based Auto-query Model Based on Sequence Annotation] (https://github.com/PaddlePaddle/models/tree/develop/neural_qa)

## 11. Image Classification

Compared with text, images can provide more vivid, easy to understand and more artistic information, which is an important source of people's transfer and exchange of information. Image classification helps to distinguish different types of images based on the semantic information of the image. It is an important problem in computer vision and is also the basis of image detection, image segmentation, object tracking, behavior analysis, and other high-level visual tasks. It is widely used in many fields like face recognition and intelligent video analysis in the field of security, traffic scene recognition in the traffic field, content-based image retrieval and album auto-categorization in the field of Internet, and image recognition in the medical field.

In this image classification task, we introduce how to train AlexNet, VGG, GoogLeNet, ResNet, Inception-v4, Inception-Resnet-V2 and Xception models. Also provided is a model conversion tool that can convert Caffe or TensorFlow trained model files to PaddlePaddle model files.

- 11.1 [Converting Caffe Model Files to PaddlePaddle Model Files] (https://github.com/PaddlePaddle/models/tree/develop/image_classification/caffe2paddle)
- 11.2 [Converting TensorFlow Model Files to PaddlePaddle Model Files] (https://github.com/PaddlePaddle/models/tree/develop/image_classification/tf2paddle)
- 11.3 [AlexNet] (https://github.com/PaddlePaddle/models/tree/develop/image_classification)
- 11.4 [VGG] (https://github.com/PaddlePaddle/models/tree/develop/image_classification)
- 11.5 [Residual Network] (https://github.com/PaddlePaddle/models/tree/develop/image_classification)
- 11.6 [Inception-v4] (https://github.com/PaddlePaddle/models/tree/develop/image_classification)
- 11.7 [Inception-Resnet-V2] (https://github.com/PaddlePaddle/models/tree/develop/image_classification)
- 11.8 [Xception] (https://github.com/PaddlePaddle/models/tree/develop/image_classification)

## 12. Target detection

The goal of the target detection task lets the computer find the location of all the targets in a given image or video frame and classifies the targets. Target detection is a very simple task for humans. However, computers can only “see” a matrix with values ​​between 0 and 255. It is difficult to solve high-level semantic concepts such as people or objects in images or video frames, and it is even more difficult to locate the objects that appear in the image. At the same time, because the target will appear in any position in the image or video frame, the shape of the target is ever-changing, and the background of input images or video frames vary greatly. There are many factors that make the target detection a challenging problem for the computer.

In this target detection task, we introduce SSD (Single Shot MultiBox Detector) method. SSD is one of the newer and better-performing detection algorithms in the field of target detection. It has the features of high detection speed and high detection accuracy.

- 12.1 [Single Shot MultiBox Detector] (https://github.com/PaddlePaddle/models/tree/develop/ssd/README.cn.md)

## 13. Scene Text recognition

STR plays a vital role in information acquisition of a scene image because the rich text information in a scene image may help us to better learn the content and meaning of the image. In addition, the development of STR has promoted the emergence of many new applications. For example, with deep learning models that automatically identify the texts in street signs, street view applications can acquire address information more accurately.

In this scene text recognition task, we introduce how to combine CNN-based image feature extraction and RNN-based sequence translation techniques to eliminate artificially defined features, avoid character segmentation, and use automatically learned image features to achieve end-to-end unconstrained character positioning and recognition.

- 13.1 [Scene Character Recognition] (https://github.com/PaddlePaddle/models/tree/develop/scene_text_recognition)

## 14. Speech Recognition

Speech Recognition Technology (Auto Speech Recognize, ASR for short) translates the content of human speech into computer-readable input, allowing the machine to “understand” human speech. This technology is used in voice assistance, voice input, voice interaction, etc.. Deep learning has achieved remarkable achievements in the field of speech recognition. The end-to-end deep learning method integrates traditional acoustic models, dictionaries, language models and other modules and no longer depends on the assumption of conditional independence of the variables in hidden Markov models. If the model becomes more concise, a neural network model will take speech features as input and directly outputs the recognized text, which has now become the most important means of speech recognition.

In the speech recognition task, we provide a complete pipeline based on the DeepSpeech2 model, including: feature extraction, data enhancement, model training, language model, decoding module, etc.. We also provide a trained model and an example. Everyone can use their own voice to have fun with speech recognition.

14.1 [Speech Recognition: DeepSpeech2] (https://github.com/PaddlePaddle/DeepSpeech)

This tutorial is authored by [PaddlePaddle] (https://github.com/PaddlePaddle/Paddle) and licensed under the [Apache-2.0] (LICENSE) license agreement.
